lib/improver/spotdata_new/spot_extraction.py

```diff
@@ -0,0 +1,247 @@
+                required spot sites.
+        Returns:
+            neighbour_cube (iris.cube.Cube):
+                A spot data cube containing the extracted diagnostic data.
+        """
+
+        neighbour_cube = build_spotdata_cube(
+            spot_values, diagnostic_cube.name(), diagnostic_cube.units,
+            neighbour_cube.coord('altitude').points,
+            neighbour_cube.coord(axis='y').points,
+            neighbour_cube.coord(axis='x').points,
+            neighbour_cube.coord('wmo_id').points)
+        return neighbour_cube
+
+    def process(self, neighbour_cube, diagnostic_cube):
+        """
+        Create a spot data cube containing diagnostic data extracted at the
+        coordinates provided by the neighbour cube.
+
+        .. See the documentation for more details about the inputs and output.
+        .. include:: extended_documentation/spot_extraction_examples.rst
+
+        Args:
+            neighbour_cube (iris.cube.Cube):
+                A cube containing information about the spot data sites and
+                their grid point neighbours.
+            diagnostic_cube (iris.cube.Cube):
+                A cube of diagnostic data from which spot data is being taken.
+        Returns:
+            spotdata_cube (iris.cube.Cube):
+                A cube containing diagnostic data for each spot site, as well
+                as information about the sites themselves.
+        """
+
+        # Check we are using a matched neighbour/diagnostic cube pair
+        if not check_cube_compatibility(self.grid_metadata_identifier,
+                                        neighbour_cube, diagnostic_cube):
+            msg = ("The grid_metadata_identifier attributes do not match "
+                   "between the neighbour cube and the diagnostic cube. "
+                   "Cubes may not be for the same grid/model.")
+            raise ValueError(msg)
+
+        coordinate_cube = self.extract_coordinates(neighbour_cube)
+
+        # Deal with leading dimensions such as thresholds, realizations, etc.
+        data_cubes = iris.cube.CubeList()
+        for cube in diagnostic_cube.slices(
+                [diagnostic_cube.coord(axis='x').name(),
+                 diagnostic_cube.coord(axis='y').name()]):
+
+            spot_values = self.extract_diagnostic_data(coordinate_cube, cube)
+            spotdata_cube = self.build_diagnostic_cube(neighbour_cube, cube,
+                                                       spot_values)
+
+            # Add scalar coordinates onto the spot cube which can be promoted
+            # to reform and leading dimensions.
+            for coord in cube.coords(dim_coords=False):
+                spotdata_cube.add_aux_coord(coord)
+            data_cubes.append(spotdata_cube)
+
+        spotdata_cube = data_cubes.merge_cube()
+
+        # Copy attributes from the diagnostic cube that describe the data's
+        # provenance.
+        spotdata_cube.attributes = diagnostic_cube.attributes
```

Comment:
I'm not sure we want to be doing this.  Is the grid metadata and title (`<Diagnostic> forecast on 2 km standard grid`) really still descriptive of a spot-extracted data cube which may then have things like lapse rate adjustments applied?

Reply:
Possibly not, but we have no better conventions at the moment. We could exclude certain bits of metadata if necessary, but it is important to know which model the data came from at stages in the processing chain before the end result; remember we will extract at every stage in the processing for verification.

Eventually I assume the final output will say ```IMPROVER level 3 UK domain``` or something similar, but we don't have that standard in place yet.

Reply:
OK.  I don't like the way we're doing this at the moment, but accept we don't have a better way just yet.  I've passed this question to @bjwheltor for the metadata guidance he's working on, so we can probably accept this for now.